File: SRC/CPCDOS EDITOR/Host/MyTopLevelComponent.vb
Imports System.ComponentModel
Imports System.ComponentModel.Design

Namespace OSMaker
    ''' <summary>
    ''' Uses the custom RootDesigner (MyRootDesigner)
    ''' </summary>
    <Designer(GetType(MyRootDesigner), GetType(IRootDesigner))>
    <Designer(GetType(ComponentDesigner))>
    Public Class MyTopLevelComponent
        Inherits Component

        Public Sub New()
        End Sub
    End Class
End Namespace ' namespace

File: SRC/CPCDOS EDITOR/Host/MyTopLevelComponent.cs
using System.ComponentModel;
using System.ComponentModel.Design;

namespace OSMaker.Host
{
    /// <summary>
    /// Uses the custom RootDesigner (MyRootDesigner)
    /// </summary>
    [Designer(typeof(MyRootDesigner), typeof(IRootDesigner))]
    [Designer(typeof(ComponentDesigner))]
    public  class MyTopLevelComponent : Component
    {
        public MyTopLevelComponent()
        {
        }
    }
} // namespace
